### PR TITLE
[JENKINS-26611] SvnTagPublisher aborted java.lang.NoSuchMethodError

### PR DESCRIPTION
**Pull request by recena**
https://issues.jenkins-ci.org/browse/JENKINS-26611

### CODE REVIEW

**Comment by recena** [comment]
@reviewbybee Could your review these changes?

**Comment by recena** [comment]
In Subversion Plugin 2.5 this method was removed (disappeared):

```java
/**
 * Gets the file that stores the revision.
 */
public static File getRevisionFile(AbstractBuild build) {
    return new File(build.getRootDir(),"revision.txt");
}
```

And it added this:

```java
/**
 * Gets the file that stores the revision.
 */
public static File getRevisionFile(Run build) {
    return new File(build.getRootDir(),"revision.txt");
}
```

There is a lot of installations of Jenkins using Subversion Plugin 2.5. This PR allows to use this plugin with Subversion Plugin 2.5

**Comment by varmenise** [comment]
:bee: looks good to me

**Comment by varmenise** [comment]
What happens if we try to run this plugin with Subversion < 2.5 and the current change? Wouldn't it be safer to add a new method and add a `@deprecated` annotation to the old method instead?

**Comment by recena** [comment]
@varmenise This bug is more difficult to explain than solve it.

1. With this PR, the user can use Subversion Plugin 2.5. They will only to update Svn Tag Plugin.
1. In Subversion 2.5.1 (I hope to get a new release as soon as possible) will exist both methods but one of them is deprecated.

**Comment by varmenise** [comment]
@recena oh right sorry 

````<groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>subversion</artifactId>
     <artifactId>subversion</artifactId>
      <version>2.5</version> ````

File: src/main/java/hudson/plugins/svn_tag/SvnTagPlugin.java [hunk 1]
```diff
@@ -233,12 +235,10 @@ static String evalGroovyExpression(Map<String, String> env, String evalText,
      */
     /*package*/
     @SuppressWarnings({"NestedAssignment"})
-    static Map<String, Long> parseRevisionFile(AbstractBuild build)
```
Comment (daniel-beck) [inline src/main/java/hudson/plugins/svn_tag/SvnTagPlugin.java]: Irrelevant formatting changes that make the PR difficult to read.

File: src/main/java/hudson/plugins/svn_tag/SvnTagPlugin.java [hunk 2]
```diff
@@ -233,12 +235,10 @@ static String evalGroovyExpression(Map<String, String> env, String evalText,
...
-            throws IOException {
-        Map<String, Long> revisions =
-                new HashMap<String, Long>(); // module -> revision
+    static Map<String, Long> parseRevisionFile(AbstractBuild build) throws IOException {
+        Map<String, Long> revisions =  new HashMap<String, Long>(); // module -> revision
         // read the revision file of the last build
-        File file = SubversionSCM.getRevisionFile(build);
+        File file = SubversionSCM.getRevisionFile((Run) build);
```
Comment (daniel-beck) [inline src/main/java/hudson/plugins/svn_tag/SvnTagPlugin.java]: I think really only this line should be changed in this file.

**Comment by daniel-beck** [comment]
> With this PR, the user can use Subversion Plugin 2.5. He will only have to update the Svn Tag Plugin.

Actually, Jenkins will update the Subversion plugin to 2.5 if this gets installed or updated, as that's a dependency now.

**Comment by stephenc** [comment]
:bug: there is a lot more changes than necessary to fix the issue. split the removal of the IDE specific files into a separate pull request and remove the irrelevant code formatting changes and I'll `s/ug/ee/`

**Pull request closed without merging**